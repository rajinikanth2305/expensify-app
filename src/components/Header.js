import React from "react";
import {NavLink} from "react-router-dom";

const Header=()=>
(
    <header>
    <h1>Tilte of tha expensify app</h1>
    <NavLink to ="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to ="/create" activeClassName="is-active">create expense page</NavLink>
    <NavLink to="/edit" activeClassName="is-active">Edit page</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>

    </header>
)
export default Header;