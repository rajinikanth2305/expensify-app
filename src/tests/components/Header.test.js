import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});




//snapshot allows to track changes in data
//renderer function takes render function and it prints the output for component
//we have to install ReactShallowRender for testing the components
//library called nyzyme  for airbnb to actually writing test cases
//step 1 to install  