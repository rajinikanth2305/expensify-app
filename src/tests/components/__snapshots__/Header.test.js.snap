// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Title of the expensify app
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    create expense page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Edit page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
